packages/components/FocusZone/macos/RCTFocusZone.mm

```diff
@@ -0,0 +1,300 @@
+};
+
+typedef enum {
+	FocusZoneActionNone,
+	FocusZoneActionTab,
+	FocusZoneActionShiftTab,
+	FocusZoneActionRightArrow,
+	FocusZoneActionLeftArrow,
+	FocusZoneActionDownArrow,
+	FocusZoneActionUpArrow,
+} FocusZoneAction;
+
+static const CGFloat FocusZoneBuffer = 3;
+
+@implementation RCTFocusZone
+
+static inline CGFloat GetDistanceBetweenPoints(NSPoint point1, NSPoint point2)
+{
+	NSPoint delta = NSMakePoint(point1.x - point2.x, point1.y - point2.y);
+	return sqrt(delta.x * delta.x + delta.y * delta.y);
+}
+
+static inline NSPoint GetCenterOfRect(NSRect rect)
+{
+	return NSMakePoint(NSMidX(rect), NSMidY(rect));
+}
+
+static inline CGFloat GetDistanceBetweenCentersOfRects(NSRect rect1, NSRect rect2)
+{
+	return GetDistanceBetweenPoints(GetCenterOfRect(rect1), GetCenterOfRect(rect2));
+}
+
+static inline CGFloat GetMinDistanceBetweenRectVerticesAndPoint(NSRect rect, NSPoint point)
+{
+	return fmin(
+		fmin(GetDistanceBetweenPoints(point, NSMakePoint(NSMinX(rect), NSMinY(rect))),
+			GetDistanceBetweenPoints(point, NSMakePoint(NSMaxX(rect), NSMaxY(rect)))),
+		fmin(GetDistanceBetweenPoints(point, NSMakePoint(NSMaxX(rect), NSMinY(rect))),
+			GetDistanceBetweenPoints(point, NSMakePoint(NSMinX(rect), NSMaxY(rect))))
+	);
+}
+
+static NSView *GetFirstResponder(NSWindow *window)
+{
+	NSResponder *responder = [window firstResponder];
+	while (responder != nil && ![responder isKindOfClass:[NSView class]])
+		responder = [responder nextResponder];
+	return [responder isKindOfClass:[NSView class]] ? (NSView *)responder : nil;
+}
+
+static void EnumerateFocusableViews(NSView *root, void (^block)(NSView *, BOOL *))
+{
+	NSMutableArray<NSView *> *queue = [NSMutableArray new] ;
+	[queue addObject:root];
+
+	while ([queue count] > 0)
+	{
+		NSView *view = [queue firstObject];
+		[queue removeObjectAtIndex:0];
+
+		if ([view canBecomeKeyView])
+		{
+			BOOL stop = NO;
+			block(view, &stop);
+			if (stop)
+			{
+				break;
+			}
+		}
+		else
+		{
+			[queue addObjectsFromArray:[view subviews]];
+		}
+	}
+}
+
+static FocusZoneAction GetActionForEvent(NSEvent *event)
+{
+	FocusZoneAction action = FocusZoneActionNone;
+
+	NSEventModifierFlags modifierFlags = [event modifierFlags]
+		& (NSEventModifierFlagShift | NSEventModifierFlagControl
+			| NSEventModifierFlagOption | NSEventModifierFlagCommand);
+
+	switch ([event keyCode])
+	{
+		case kVK_UpArrow:
+			action = FocusZoneActionUpArrow;
+			break;
+
+		case kVK_DownArrow:
+			action = FocusZoneActionDownArrow;
+			break;
+
+		case kVK_LeftArrow:
+			action = FocusZoneActionLeftArrow;
+			break;
+
+		case kVK_RightArrow:
+			action = FocusZoneActionRightArrow;
+			break;
+
+		case kVK_Tab:
+			if (modifierFlags == 0)
+			{
+				action = FocusZoneActionTab;
+				break;
+			}
+			else if (modifierFlags == NSEventModifierFlagShift)
+			{
+				action = FocusZoneActionShiftTab;
+				break;
+			}
+	}
+
+	return action;
+}
+
+- (NSView *)getNextView:(BOOL)next toFocusInRowOrColumn:(BOOL)horizontal
```

Comment:
unless there is a reason why we need to have "get" in the prefix of this function name it doesn't seem very objc style function naming. applies for other functions too

Reply:
"tofocusinroworcolumn" doesn't make much sense. or Hoirzontal doesn't really mean boolean